x-pack/plugins/synthetics/public/apps/synthetics/components/test_run_details/components/step_number_nav.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface Props {
+  stepIndex: number;
+  totalSteps: number;
+  hasPreviousStep: boolean;
+  hasNextStep: boolean;
+  handlePreviousStep: () => void;
+  handleNextStep: () => void;
+}
+
+export const StepNumberNav = ({
+  stepIndex,
+  totalSteps,
+  handleNextStep,
+  handlePreviousStep,
+  hasNextStep,
+  hasPreviousStep,
```

Comment:
I think this component can infer this information from `stepIndex` and `totalSteps`.